### PR TITLE
answers for lesson 1

### PR DESCRIPTION
**Pull request by iArtemv**

### CODE REVIEW

File: 01-node-tutorial/answers/02-globals.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+console.log("Dir: " + __dirname);
```
Comment (rohwedernt) [inline 01-node-tutorial/answers/02-globals.js]: This is great! Nothing wrong with this syntax whatsoever but just something to chew on here - console logs can take all kinds of args.- objects, arrays, functions or multiple variables. I sometimes find it easier in a case like this to just psas 2 args like this `console.log("Dir: ", __dirname)`. (all just a matter of preference though)

File: 01-node-tutorial/answers/01-intro.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const txt = "Hello World!";
+
+if (txt.length > 0) {
+    console.log(txt.toUpperCase());
+}
```
Comment (rohwedernt) [inline 01-node-tutorial/answers/01-intro.js]: You may be aware of this but all of these red circles just mean git likes there to be an empty line at the end of the file.

File: 01-node-tutorial/answers/08-os-module.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+console.log(`The System uptime is ${os.uptime} seconds`);
+
+const currentOS = {
+    name: os.type(),
+    release: os.release(),
+    mem: os.totalmem(),
+    freeMem: os.freemem(),
+};
+
+console.log(currentOS);
```
Comment (rohwedernt) [inline 01-node-tutorial/answers/08-os-module.js]: This is great, well done.

File: 01-node-tutorial/answers/09-path-module.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const path = require('path');
+
+const file = path.join('node-express-course', '01-node-tutorial','answers','09-path-module.js');
+
+//console.log(file);
```
Comment (rohwedernt) [inline 01-node-tutorial/answers/09-path-module.js]: Super minor but folks will generally call out commented out console logs in PRs to be removed. It is common to need to log something while developing but if it is not needed long-term it is generally good practice to remove it.

File: 01-node-tutorial/answers/10-fs-sync.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+const fs = require('fs');
+const path = require('path');
+
+const filePath = path.join(__dirname, './temporary', 'fileA.txt');
+
+if (!fs.existsSync(path.join(__dirname, './temporary'))) {
```
Comment (rohwedernt) [inline 01-node-tutorial/answers/10-fs-sync.js:6]: great job implementing this guard here.

**Comment by rohwedernt** [comment]
Thanks Artem! This is great. The only thing that appears to be missing here are the updates to `prompter.js`. Were you able to modify that?

**Comment by rohwedernt** [comment]
Looks like you felt pretty good about these tasks. In the future though, always feel free to post comments on your own PRs with specific questions you may have had.

File: 01-node-tutorial/answers/02-globals.js (hunk 1 above)
Comment (iArtemv) [reply 01-node-tutorial/answers/02-globals.js]: This way is better than string concatenation. Thank you!

File: 01-node-tutorial/answers/01-intro.js (hunk 2 above)
Comment (iArtemv) [reply 01-node-tutorial/answers/01-intro.js]: I didn't pay attention to this before, but it really will make the file more readable. Thanks

**Comment by iArtemv** [comment]
@rohwedernt For some reason I overlooked the task about prompter.js. I have completed the tasks and will upload them with corrections according to the CR. (Thank you very much for your useful comments!)

As I completed the tasks, I had a couple of questions.
1 - what is the difference between different imports "require" and "import"? - I seem to have figured out this issue, but not completely yet.
2 - noticed that there is a strict difference between a pattern with back quotes and a string with regular quotes, that is, even if a variable is specified in the pattern, its value must be specified as a pattern - and not as just a string. I haven’t fully understood this issue yet.

**Comment by rohwedernt** [comment]
Hey @iArtemv! No problem, thanks for the quick response. 

1 - require and import are the _import_ or _binding_ mechanisms for 2 different module systems. require comes from CommonJS modules and import comes from EcmaScript (ES6) modules. There will be plenty of time to dig into what this means but for now it's important to note that these don't behave exactly the same way and import particularly can require some setup with node. Take a look [here](https://www.scaler.com/topics/nodejs/require-vs-import-nodejs/) for a pretty succinct overview (or scroll directly down the table labeled "Difference between Require and Import" near the bottom). 
2 - I think what you're asking about here is something call template literals (or template strings) in javascript. This can be a super powerful bit of syntax for inserting data dynamically into a string. Regular quotes on the other hand are just plain old strings and don't have the same functionality. More reading [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) if you're interested.

**Review by rohwedernt — APPROVED** [review]

**Pull request merged**